### PR TITLE
feat(ui5-page): Initial implementation

### PR DESCRIPTION
**Pull request by ivoplashkov**
FIXES: #2433

### CODE REVIEW

File: packages/fiori/src/Page.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+ * @public
+ */
+const metadata = {
+	tag: "ui5-page",
+	managedSlots: true,
+	languageAware: true,
+	properties: /** @lends sap.ui.webcomponents.fiori.Page.prototype */ {
+
+		/**
+         * This property is used to set the background color of the <code>ui-page</code>. When a list is placed inside the page, the value "List" should be used to display a gray background. "Standard", with the default background color, is used if not specified.
```
Comment (fifoosid) [inline packages/fiori/src/Page.js:23]: <code>ui5-page</code>

File: packages/fiori/src/Page.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+		},
+
+		/**
+         * Decides whether the footer would be visible.
+         *
+		 * @type {Boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		hideFooter: {
```
Comment (fifoosid) [inline packages/fiori/src/Page.js:72]: This property is not needed. If someone wants to hide the footer, they can just remove it from the DOM

File: packages/fiori/src/Page.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+	static get template() {
+		return PageTemplate;
+	}
+
+	static async onDefine() {
+		await fetchI18nBundle("@ui5/webcomponents");
+	}
+
+	get _contentBottom() {
+		return !this.floatingFooter && !this.hideFooter ? "2.75rem" : "0";
```
Comment (fifoosid) [inline packages/fiori/src/Page.js:174]: These styles could be moved to a separate CSS class, which can be applied based on the same logic in the get classes method.

File: packages/fiori/src/Page.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+	}
+
+	get _contentPaddingBottom() {
+		return this.floatingFooter && !this.hideFooter ? "3.5rem" : "0";
```
Comment (fifoosid) [inline packages/fiori/src/Page.js:178]: Same here

File: packages/fiori/src/Page.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+				"ui5-page-show-floating-footer": !this.hideFooter && this.floatingFooter,
+				"ui5-page-hide-floating-footer": this.hideFooter && this.floatingFooter,
+			},
+		};
+	}
+
+	get styles() {
+		return {
+			content: {
+				"padding-bottom": this.footer.length && this._contentPaddingBottom,
```
Comment (fifoosid) [inline packages/fiori/src/Page.js:194]: If the styles are moved to a CSS class, this logic would not be needed

File: packages/fiori/src/Page.js [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+				"bottom": this.footer.length && this._contentBottom,
+				"overflow": this.disableScrolling ? "hidden" : "auto",
```
Comment (fifoosid) [inline packages/fiori/src/Page.js:196]: You can set one of the values as default. You can create then a CSS selector with the attribute that corresponds to the property and override the default value with stronger CSS selector

File: packages/fiori/src/Page.js [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+			},
+			footer: {
+				"display": this.hideFooter && !this.floatingFooter ? "none" : "block",
```
Comment (fifoosid) [inline packages/fiori/src/Page.js:199]: Same here. This way you will get rid of the get styles method

File: packages/fiori/src/Page.js [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents-fiori/dist/Page.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.fiori.Page
+ * @extends UI5Element
+ * @tagname ui5-page
+ * @since 1.0.0-rc.11
```
Comment (fifoosid) [inline packages/fiori/src/Page.js:144]: rc.12

**Review by fifoosid — CHANGES_REQUESTED** [review]
This review is related to the code of the change, not so much to the look & behaviour of the component. The change looks good to me, there are just a few small comments to be addressed.

**Comment by ivoplashkov** [comment]
> This review is related to the code of the change, not so much to the look & behaviour of the component. The change looks good to me, there are just a few small comments to be addressed.

Thanks for the review! The hideFooter property should stay as this has been already discussed, because there are animations attached to the floating footer that should be executed. Probably the "disableScrolling" property is redundant as it applies a single CSS style. The applied styles could not be moved to the CSS, because we must ensure that there is a footer in order to apply the CSS class, which can be checked only if I introduce a private property, but I don't think it's the better choice here.
Other comments have been resolved.

File: packages/fiori/src/themes/Page.css [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    animation: bounceShow 0.35s forwards ease-in-out;
+}
+
+:host([floating-footer][hide-footer]) .ui5-page-footer-root {
+    animation: bounceHide 0.35s forwards ease-in-out;
+}
+
+/* BackgroundDesign */
+:host([backgroundDesign="Standard"]) .ui5-page-root {
+	background-color: var(--sapBackgroundColor);
```
Comment (ilhan007) [inline packages/fiori/src/themes/Page.css:55]: Is it possible to change the content background easily (as with the CSS below).

If you set these backgrounds on the host and inherit the background in .ui5-page-root element, the users would be able to easily set a custom background like:
```css
ui5-page {
background: red;
}
```

Comment (ivoplashkov) [reply packages/fiori/src/themes/Page.css:56]: Thanks, done.

File: packages/fiori/test/specs/Page.spec.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+const assert = require('chai').assert;
+
+describe("Page general interaction", () => {
+	browser.url("http://localhost:8081/test-resources/pages/Page.html");
+
+	it("tests initial rendering", () => {
+        const page = browser.$("#page");
+
+        assert.ok(page.isExisting(), "The component should have a shadow root.");
+        assert.ok(page.shadow$(".ui5-page-header-root"), "Header should be rendered.");
```
Comment (ilhan007) [inline packages/fiori/test/specs/Page.spec.js:10]: This pattern below will be always truthy - isExisting() is the way to check if DOM element is rendered, like `page.isExisting()`.

```js
page.shadow$(".ui5-page-header-root")
```

Comment (ilhan007) [inline packages/fiori/test/specs/Page.spec.js:10]: This pattern below will be always truthy, becasue wdio will return a wrapper object always no matter the DOM element is rendered or not.

The isExisting() is the way to check if DOM element is rendered, like `page.isExisting()`.

```js
page.shadow$(".ui5-page-header-root")
```

**Review by ilhan007 — APPROVED** [review]

**Pull request merged**